Hyperion: Add brightness, HDMI and effect support

starkillerOG opened the pull request:
## Description:
- added brightness support to dim the hyperion light
- changed the "OFF" command to set the color to [0,0,0] after clearing all priorities.
  This is neccesary to keep the light turned off when an HDMI grabber is used for ambilight with hyperion.
- added HDMI ambilight mode recognition and control.
  by setting the "hdmi_priority" in your "configuration.yaml" file (defaults to 880), home assistant will now be able to recognize when the hyperion light is in HDMI ambilight mode and will change its icon to an HDMI symbol and set the status to ON.
Switching the hyperion light to HDMI ambilight mode can be done through the effect option (clears all priorities such that the HDMI grabber remains).
- added effect support for the default effects of hyperion, a custom list can be defined in the "configuration.yaml" file by using the "effect_list" option.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#4376

## Example entry for `configuration.yaml` (if applicable):
```yaml
 - platform: hyperion
   host: IP_ADRESS
   port: 19444
   name: hyperion light
   priority: 100
   hdmi_priority: 880
   default_color: [0,0,255]
   effect_list: ['HDMI', 'Blue mood blobs']
```

## Checklist:
  - [ ] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support

- added brightness support to dim the hyperion light
- changed the "OFF" command to set the color to [0,0,0] after clearing all priorities.
  This is neccesary to keep the light turned off when an HDMI grabber is used for ambilight with hyperion.
- added HDMI ambilight mode recognition and control.
  by setting the "hdmi_priority" in your "configuration.yaml" file (defaults to 880), home assistant will now be able to recognize when the hyperion light is in HDMI ambilight mode and will change its icon to an HDMI symbol and set the status to ON.
Switching the hyperion light to HDMI ambilight mode can be done through the effect option (clears all priorities such that the HDMI grabber remains).
- added effect support for the default effects of hyperion, a custom list can be defined in the "configuration.yaml" file by using the "effect_list" option.

homeassistant commented:
Hi @starkillerOG, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#11532).

Once you do that we will be able to review and accept this pull request.

Thanks!

tschmidty69 commented:
You need to redo this pull request and compare against the dev branch (https://home-assistant.io/developers/development_submitting/). 

Also a good idea to look here so you can do your lint checks locally to avoid having to redo: https://home-assistant.io/developers/development_testing/

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support

- added brightness support to dim the hyperion light
- changed the "OFF" command to set the color to [0,0,0] after clearing all priorities.
  This is neccesary to keep the light turned off when an HDMI grabber is used for ambilight with hyperion.
- added HDMI ambilight mode recognition and control.
  by setting the "hdmi_priority" in your "configuration.yaml" file (defaults to 880), home assistant will now be able to recognize when the hyperion light is in HDMI ambilight mode and will change its icon to an HDMI symbol and set the status to ON.
Switching the hyperion light to HDMI ambilight mode can be done through the effect option (clears all priorities such that the HDMI grabber remains).
- added effect support for the default effects of hyperion, a custom list can be defined in the "configuration.yaml" file by using the "effect_list" option.
- fixed some style issues with too long lines

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support 

 - fixed some more indentation style issues

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support 

- yet more fixed visuel indent issues

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support 

- more visuel indents

starkillerOG commented:
Alright, I changed the base brench to the dev version and I fixed all those anoying visual indent issues.
Can this now be merged, or is there anything else I need to do?

starkillerOG commented:
@tschmidty69 How can I remove the red "merging to master" label, since I have changed the base branch to dev?

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support

- fixed invalid variable "A"

tschmidty69 commented:
Unfortunately I don't have write access so I can't . I think you probably need to close this and open a new pull request.

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support

- remove unnececary brackets
- specify specific exceptions

starkillerOG commented:
@tschmidty69 Alright I will close and open a new one tommorow.
First I will get rid of those last few annoying issues with the checks.
Hopfully this last commit will pass all tests.

starkillerOG commented:
Closing and reopening with the dev branch as base

starkillerOG commented:
New pull request: #11532
